src/goto-instrument/contracts/contracts.cpp

```diff
@@ -853,18 +856,16 @@ bool code_contractst::check_frame_conditions_function(const irep_idt &function)
     function,
     symbol_table);
 
-  // Adds formal parameters to write set
+  // fetch local statics of all subfunctions in the symbol_table
+  find_static_locals(goto_functions, function, assigns);
+
+  // Full inlining of the function body
+  goto_function_inline(goto_functions, function, ns, log.get_message_handler());
```

Comment:
I think the comment could be more useful if it explained _why_ full inlining was necessary. (Also, will this work with recursive functions?)

Reply:
the reason behind inlining is that in the function `f` we are instrumenting ,  a same function `g` can be called in several locations under different write sets. 
Without inlining, for each call site we would have to create a copy of g and instrument for the write set specific to this call site.
Inlining allows us to avoid having to duplicate g and instrument it specifically. 

As for inlining recursive functions, I'll implement a check to detect them and error out during inlining